Pulse draw fix

Dpbm opened the pull request:
<!--
⚠️ If you do not respect this template, your pull request will be closed.
⚠️ Your pull request title should be short detailed and understandable for all.
⚠️ Also, please add a release note file using reno if the change needs to be
  documented in the release notes.
⚠️ If your pull request fixes an open issue, please link to the issue.

- [x] I have added the tests to cover my changes.
- [x] I have updated the documentation accordingly.
- [x] I have read the CONTRIBUTING document.
-->

### Summary

A check for the `.name()` function was added to mitigate `TypeError` in `OpenPulseBackendInfo.create_from_backend` method. It happens when a backend based on `BackendV2`, was passed to it.

### Details and comments

The problem occurs because the `BackendV2` interface doesn't have a `.name()` method, but a `.name` attribute, then when the `.create_from_backend` calls for it, a `TypeError` raises.
Now a `try-except` block was added to check it
```python
try:
   name = backend.name()
except TypeError:
   name = configuration.backend_name
```

From issue #10993.

Dpbm committed:
added test for BackendV2

Dpbm committed:
added test for BackendV1

Dpbm committed:
fixed pulse_v2 device_info get backend name

Dpbm committed:
pass lint

Dpbm committed:
added releasenote

Dpbm committed:
Merge branch 'Qiskit:main' into pulse_draw_fix

jakelishman reviewed (commented):
I mentioned this in #10993, but I'm sorry: this isn't a proper fix.  This happens to work for the backend given in that issue, but say `backend.configuration()` is not a method that exists on the `BackendV2` interface, and so this only works for the IBM backends that happen to fulfil much of both of the interfaces.  A full fix of these needs to properly distinguish `BackendV1` and `BackendV2` inside `create_from_backend`, and only use methods / attributes defined by those interfaces.

Dpbm commented:
Oh, sorry about that! My mind has been full these days, and I ended skimming over your message without reading it properly.

---

Now I'm working to fix it properly, but just one question. When I read your reply, just came into my mind creating some functions to return the correct values for each backend version, but after some research I came across with `BackendV2Converter`.
Is that bad using it to convert the input backend, case it's using `BackendV1`, to set `BackendV2` as the default version in `create_from_backend`? Or the first approach is better?

Dpbm committed:
Merge branch 'Qiskit:main' into pulse_draw_fix

Dpbm committed:
adding backends adapters

Dpbm committed:
updated test for backendV1

Dpbm committed:
updated BackendV2 adapter

Dpbm committed:
updated tests

Dpbm committed:
added get u_channel_lo from target

Dpbm committed:
added test for BackendV1 - invalid backend name

Dpbm committed:
added test for BackendV1 - invalid backend defaults

Dpbm committed:
added test for BackendV1 - invalid backend dt

Dpbm committed:
updated backendV1 adapter tests

Dpbm committed:
updated backendV2 adapter tests

Dpbm committed:
cleaned backendV1 adapter tests

Dpbm committed:
cleaned backendV2 adapter tests

Dpbm committed:
added raise_attribute_doesnt_exist

Dpbm committed:
added tests for get_backend_adapter

Dpbm committed:
added tests for create backend

Dpbm committed:
changed tests for get_backend_adapter

Dpbm committed:
did lint

Dpbm committed:
updated get u_channel_lo

Dpbm committed:
updated get u_channel_lo approach

Dpbm committed:
updated release note

Dpbm committed:
Merge branch 'Qiskit:main' into pulse_draw_fix

Dpbm commented:
I think I've done the fix. But there're two things that I'm not really sure about it.

1. in the adapter for the BackendV2, to get the `u_channel_lo` I'm looking inside `.target`, then, if it's not there, I try to get  from `backend.u_channel_lo`. I don't really know if it's a good idea.
2. I added a bunch of tests to check every inch in my code (including, sorry about the huge amount of code for tests), but I don't know if they are really good for this case. Wether you think that they are not sufficient or they are useless, I'll be pleasure to learn and update my code.    

---

Besides that, I've added two adapters, one for each backend version, and now I'm checking what version is and dispatching the correct adapter.

jakelishman commented:
Closing as stale, since these were removed in 2.0
